sdk/python/lib/pulumi/runtime/resource.py

```diff
@@ -661,15 +657,37 @@ def convert_providers(
     return result
 
 
-async def _resolve_depends_on_urns(options: 'ResourceOptions') -> List[str]:
+async def _add_dependency(deps: Set[str], res: 'Resource'):
+    from .. import ComponentResource # pylint: disable=import-outside-toplevel
+
+    if isinstance(res, ComponentResource):
```

Comment:
Should we include a comment along the lines of the following somewhere in the Python SDK?

https://github.com/pulumi/pulumi/blob/e567b4762fe0dbb88ec35db2f3677a95cfdd715b/sdk/nodejs/runtime/resource.ts#L600-L627

Reply:
Yes, that's a good idea.